frontend/src/components/eMIB/Inbox.jsx

```diff
@@ -24,51 +22,62 @@ const styles = {
   }
 };
 
+function initializeFalseArray(length) {
+  var arr = [];
+  for (var i = 0; i < length; i++) {
+    arr.push(false);
+  }
+  return arr;
+}
+
 class Inbox extends Component {
   static propTypes = {
-    inboxSpecs: PropTypes.array.isRequired
+    inboxLength: PropTypes.number.isRequired
   };
 
   state = {
-    currentEmail: 0
+    currentEmail: 0,
+    emailRead: initializeFalseArray(this.props.inboxLength),
+    emailReplies: initializeFalseArray(this.props.inboxLength)
   };
 
   changeEmail = index => {
-    this.setState({ currentEmail: index - 1 });
+    var array = this.state.emailRead;
```

Comment:
This doesn't actually make a copy of the state. And you shouldn't being using var in ES6 JS, you should be using let and const.

Here's another way to write this function:
```
let emailsRead = Array.from(this.state.emailRead);
emailsRead[index] = true;
// The previous "current email" is marked as open.
emailsRead[this.state.currentEmail] = true;
this.setState({ currentEmail: index, emailRead: emailsRead });
```

Reply:
fixed in 77d9bc6